### PR TITLE
[Issue 5] Publishing snapshot artifacts to JCenter

### PR DESCRIPTION
**Pull request by yangb8**
This is a similar code change of https://github.com/pravega/flink-connectors/pull/129

Test
- docker run --name artifactory -d -p 8081:8081 docker.bintray.io/jfrog/artifactory-oss
- ./gradlew
 publishToRepo -PpublishUrl=http://localhost:8081/artifactory/example-repo-local/ -PpublishUsername=admin -PpublishPassword=password
- http://localhost:8081/artifactory/example-repo-local/io/pravega/hadoop-connectors/
hadoop-connectors-0.3.0-16.4331d47-20180523.183352-1-javadoc.jar   23-May-2018 18:33  31.84 KB
hadoop-connectors-0.3.0-16.4331d47-20180523.183352-1-sources.jar   23-May-2018 18:33  6.76 KB
hadoop-connectors-0.3.0-16.4331d47-20180523.183352-1.jar           23-May-2018 18:33  22.43 MB
hadoop-connectors-0.3.0-16.4331d47-20180523.183352-1.pom           23-May-2018 18:33  1.55 KB
maven-metadata.xml                                                 23-May-2018 18:33  374 bytes

Also uploaded to jfrog

https://oss.jfrog.org/artifactory/jfrog-dependencies/io/pravega/pravega-connectors-hadoop/0.3.0-18.a042f0c-SNAPSHOT/
../
maven-metadata.xml                                                         24-May-2018 21:40  1.23 KB
pravega-connectors-hadoop-0.3.0-18.a042f0c-20180524.214012-1-javadoc.jar   24-May-2018 21:40  33.24 KB
pravega-connectors-hadoop-0.3.0-18.a042f0c-20180524.214012-1-sources.jar   24-May-2018 21:40  6.79 KB
pravega-connectors-hadoop-0.3.0-18.a042f0c-20180524.214012-1.jar           24-May-2018 21:40  22.43 MB
pravega-connectors-hadoop-0.3.0-18.a042f0c-20180524.214012-1.pom           24-May-2018 21:40  1.55 KB

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+  - secure: "VH9F5YUSBh16A3gmTq/vE5Qt/yl+p8+TwrOIeELF7YvYS6915+Ju0BiLwq6EvJ2eZf3OZi5bdmXgdiuUHeyYtHCv4LK/RhClOeF0NtqlltXPUPKcoA+urgsT+UKQ7uSFgMaHhG+jDDpueJjKqNwmuxBpe4AcumCilmF/tM0EZucGD40JjoAJBQydLygmmoucp5KMNAr+xrp3JNC10T3HvtwTwnxwk00deEiHY5BctyIaS63hqFml1pIF69bL7+lsVfaw8IvtFBypGvcJPiUSbQVlj9zSU7ur+q7be/KUtvL99X+ZgyoGgrGmrE/qKubb6S9G8YtMWSy57CE1RHcZNjXMOIQ0PJlRMLpzH1KZQLHn3Pnu/vN/tAkSCgLExVo3DOteaanT0ve+Nxp8feNORb0OtKvlcpLgB7ke+VziZRx0Rlhvao2DmjIcoe9Re8uUszLyQf+c6rpiiCe9L1N6V+6WBSB4nYEe/s6kQ/dsu3St2ttePKXdsV2QhhrsupBv2qk8LL0AFJT9ECqAqRHQkVCB2EJLd/zRNdSSwJ2xWhJ42t6PKSfLrPXlLQKN1GCLSp+6io/6p0RD2dvVrwY5gtD309krKOdD9nqGvbZlIm1oq/zKuH0BoKN5JQ4qCI8ju9J0RjMD53vtQnMbD7mKVE0iaS1Q8Rk+2fMDZGMEkLo="
+
```
Comment (vijikarthi) [inline .travis.yml:9]: I assume the secure tokens are generated from `pravega/hadoop-connectors` namespace?

Comment (yangb8) [reply .travis.yml]: yes, they're the ones you generated, and I've verified publication on jfrog https://oss.jfrog.org/artifactory/jfrog-dependencies/io/pravega/pravega-connectors-hadoop/0.3.0-18.a042f0c-SNAPSHOT/

Comment (yangb8) [reply .travis.yml:9]: yes, they're the ones you generated, and I've verified publication on jfrog https://oss.jfrog.org/artifactory/jfrog-dependencies/io/pravega/pravega-connectors-hadoop/0.3.0-18.a042f0c-SNAPSHOT/

**Comment by yangb8** [comment]
any other comments?

**Review by vijikarthi — APPROVED** [review]

File: gradle/bintray.gradle [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    sign = true
+                }
+            }
+        }
+        publications = ['bintrayPublication']
+    }
+}
+
+def pomConfig = {
+    name "Pravega Haddopo Connectors"
```
Comment (fpj) [inline gradle/bintray.gradle]: Typo in Hadoop.

File: gradle/bintray.gradle [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                            dependencyNode.appendNode('groupId', dep.group)
+                            dependencyNode.appendNode('artifactId', dep.name)
+                            dependencyNode.appendNode('version', dep.version)
+                            dependencyNode.appendNode('scope', 'provided')
+                        }
+                    }
+
+                    root.children().last() + pomConfig
+                }
+
```
Comment (fpj) [inline gradle/bintray.gradle]: Remove new line.

File: gradle/bintray.gradle [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        user = publishUsername
+        key = publishPassword
+        pkg {
+            repo = "${bintrayRepoName}"
+            name = "${bintrayPackageName}"
+            userOrg = "${bintrayOrgName}"
+            licenses = ['Apache-2.0']
+            publish = "${autoPublish}"
+            vcsUrl = 'https://github.com/pravega/hadoop-connectors.git'
+            version {
```
Comment (fpj) [inline gradle/bintray.gradle:21]: One clarification, is there a relationship between this version here and the one here:

https://github.com/pravega/hadoop-connectors/pull/32/files#diff-c197962302397baf3a4cc36463dce5eaR111

**Review by fpj — CHANGES_REQUESTED** [review]

File: gradle/bintray.gradle (hunk 2 above)
Comment (yangb8) [reply gradle/bintray.gradle]: accepted

File: gradle/bintray.gradle (hunk 3 above)
Comment (yangb8) [reply gradle/bintray.gradle]: accepted

File: gradle/bintray.gradle (hunk 4 above)
Comment (yangb8) [reply gradle/bintray.gradle:21]: no, there is no relationship between them.

**Review by EronWright — APPROVED** [review]
This looks great

**Pull request merged**